backend/shared_utilities/decorators.py

```diff
@@ -100,3 +112,40 @@ def wrapper(*args, **kwargs):
 		return view_func(*args, **kwargs)
 	return wrapper
 
+def check_that_valid_tournament_request(tournament_states: Iterable[TournamentState]):
+	'''Check that a request that modifies tournament resource is valid.
+
+	The user making the request must be the tournament host.
+	The tournament state must in [tournament_states].
+	The tournament cannot be expired. If it is, set state to aborted.
+	'''
+	def decorator(view_func):
+		@wraps(view_func)
+		def wrapper(*args, **kwargs):
+			request = args[0] if args else None
+
+			if request.user.is_superuser:
+				return view_func(*args, **kwargs)
+
+			tournament_id = kwargs.get('tournament_id', None)
+			tournament = Tournament.objects.filter(id=tournament_id).first()
+			if not tournament:
+				return HttpResponseNotFound(f"Tournament {tournament_id} does not exist.")
+
+			if request.user != tournament.host_user:
+				return HttpResponseForbidden("You are not authorized to modify this resource.")
+
+			if tournament.state not in tournament_states:
+				return HttpResponseForbidden(f"Cannot modify this resource because tournament state is not in '{tournament_states}'.")
+			
+			try:
+				TournamentInProgressManager.make_sure_tournament_has_not_expired(tournament)
```

Comment:
away with "make_sure". is_tournament_active/ check_tournament_active/

Reply:
A general convention is usually indeed:
- if its a if statement based boolean function, starts with 'is' or 'are' and returns if true or not
- if a try catch, starts with 'try'
- if its a if statement based function that does something in a true / false case -> 'delete_if_' etc..

In general for boolean functions there is a rule in Clean Code by Uncle Bob: _"Avoid negative name for boolean function"_, its weird to assume a true result from a function called 'does_NOT_fulfil_the_if'. Refer to for example: https://levelup.gitconnected.com/knot-of-nots-avoiding-negative-names-for-boolean-methods-641896a94a42